### ISSUE TITLE
[MAYA-125991] Unable to switch variants when Session Layer targeted

### ISSUE THREAD

**Issue by tanant** [Autodesk/maya-usd#2693]
**Describe the bug**
When the Session layer edit target is set, and a user tries to change variants it looks like this is blocked out with the following error message:

```
# Error: RuntimeError: file C:\Program Files\Autodesk\Maya2023\Python\lib\site-packages\maya\
app\ufe\outlinerSupport.py line 26: Cannot set variant set modelingVariant to variant 
ChairA on [ChairB_2] because it is defined inside the variant composition arc 
[Kitchen_set.usd],[Chair.usd],[Chair.usd],[Chair_payload.usd],[Chair.geom.usd],[Chair.geom.usd].
```
(manual linebreaks added so you can read it, it's just one giant line)

**Steps to reproduce**
1. Create a stage from Kitchen_Set.usd (the pixar kitchen set)
2. Open the Layer Editor.
3. Set the Session Layer as the target.
4.  Expand to ChairB_1 in the Outliner.
5. Right Click and change the Variant.

This is a regression that doesn't occur in 0.19.0

**Expected behavior**
I expect the variant selection to be successful, and the edit (to set, in this case, the `modellingVariant`) to be made in the session layer.

**Comment by msercheli** [comment]
I am experiencing the same issue (it was not there in 0.19.0).
On top of that, if the root layer is set (in the Layer editor), Maya 2022.3 crashes with the following error:

Could not resolve material <>

Stack trace:
  wcslen
  awString::IString::assign(wchar_t const*)
  Tstring::set(wchar_t const*)
  Autodesk::Maya::OpenMaya20230000::MHWRender::MRenderItem::name() const
  pxrInternal_v0_22__pxrReserved__::HdVP2DrawItem::~HdVP2DrawItem()
  pxrInternal_v0_22__pxrReserved__::HdVP2DrawItem::~HdVP2DrawItem()
  pxrInternal_v0_22__pxrReserved__::HdRepr::~HdRepr()
  pxrInternal_v0_22__pxrReserved__::HdRprim::~HdRprim()
  pxrInternal_v0_22__pxrReserved__::HdVP2Mesh::~HdVP2Mesh()
  pxrInternal_v0_22__pxrReserved__::HdRenderIndex::_RemoveRprim(pxrInternal_v0_22__pxrReserved__::SdfPath const&)
  pxrInternal_v0_22__pxrReserved__::HdSceneIndexAdapterSceneDelegate::PrimsRemoved(pxrInternal_v0_22__pxrReserved__::HdSceneIndexBase const&, pxrInternal_v0_22__pxrReserved__::TfSmallVector<pxrInternal_v0_22__pxrReserved__::HdSceneIndexObserver::RemovedPrimEntry, 16u> const&)
  pxrInternal_v0_22__pxrReserved__::HdSceneIndexBase::_SendPrimsRemoved(pxrInternal_v0_22__pxrReserved__::TfSmallVector<pxrInternal_v0_22__pxrReserved__::HdSceneIndexObserver::RemovedPrimEntry, 16u> const&)
  pxrInternal_v0_22__pxrReserved__::HdMergingSceneIndex::_PrimsRemoved(pxrInternal_v0_22__pxrReserved__::HdSceneIndexBase const&, pxrInternal_v0_22__pxrReserved__::TfSmallVector<pxrInternal_v0_22__pxrReserved__::HdSceneIndexObserver::RemovedPrimEntry, 16u> const&)
  pxrInternal_v0_22__pxrReserved__::HdSceneIndexBase::_SendPrimsRemoved(pxrInternal_v0_22__pxrReserved__::TfSmallVector<pxrInternal_v0_22__pxrReserved__::HdSceneIndexObserver::RemovedPrimEntry, 16u> const&)
  pxrInternal_v0_22__pxrReserved__::HdNoticeBatchingSceneIndex::_PrimsRemoved(pxrInternal_v0_22__pxrReserved__::HdSceneIndexBase const&, pxrInternal_v0_22__pxrReserved__::TfSmallVector<pxrInternal_v0_22__pxrReserved__::HdSceneIndexObserver::RemovedPrimEntry, 16u> const&)
  pxrInternal_v0_22__pxrReserved__::HdSceneIndexBase::_SendPrimsRemoved(pxrInternal_v0_22__pxrReserved__::TfSmallVector<pxrInternal_v0_22__pxrReserved__::HdSceneIndexObserver::RemovedPrimEntry, 16u> const&)
  pxrInternal_v0_22__pxrReserved__::HdRetainedSceneIndex::RemovePrims(pxrInternal_v0_22__pxrReserved__::TfSmallVector<pxrInternal_v0_22__pxrReserved__::HdSceneIndexObserver::RemovedPrimEntry, 16u> const&)
  pxrInternal_v0_22__pxrReserved__::HdRenderIndex::RemoveRprim(pxrInternal_v0_22__pxrReserved__::SdfPath const&)
  pxrInternal_v0_22__pxrReserved__::UsdImagingIndexProxy::_ProcessRemovals()
  pxrInternal_v0_22__pxrReserved__::UsdImagingDelegate::ApplyPendingUpdates()
  pxrInternal_v0_22__pxrReserved__::UsdImagingDelegate::SetTime(pxrInternal_v0_22__pxrReserved__::UsdTimeCode)
  pxrInternal_v0_22__pxrReserved__::ProxyRenderDelegate::_UpdateSceneDelegate()
  pxrInternal_v0_22__pxrReserved__::ProxyRenderDelegate::update(Autodesk::Maya::OpenMaya20230000::MHWRender::MSubSceneContainer&, Autodesk::Maya::OpenMaya20230000::MHWRender::MFrameContext const&)
  /mnt/rez/softwares/binaries/maya/2023.2/maya2023/lib/libOpenMayaRender.so(+0xe26df) [0x7fcbb81136df]
  OGSSubSceneItem::update()
  OGSMayaSubSceneManager::UpdateAllSubScenes()
  /mnt/rez/softwares/binaries/maya/2023.2/maya2023/lib/libOGSMayaBridge.so(+0x1f8705) [0x7fcc06368705]
  /mnt/rez/softwares/binaries/maya/2023.2/maya2023/lib/libOGSMayaBridge.so(+0x1f8038) [0x7fcc06368038]
  OGSApplicationBridge::UpdateScene(OGSApplicationUpdateCallback&)
  OGSMayaRenderer::updateScene(OGSApplicationUpdateCallback&, bool*)
  OGSMayaBaseRenderer::performSceneRender(OGSMayaSceneOperation*, unsigned long, bool*)
  /mnt/rez/softwares/binaries/maya/2023.2/maya2023/lib/libOGSMayaBridge.so(+0x1eeacb) [0x7fcc0635eacb]
  /mnt/rez/softwares/binaries/maya/2023.2/maya2023/lib/libOGSMayaBridge.so(+0x1f5349) [0x7fcc06365349]
  TidleRefreshCmd::refreshOGSRender(T3dView*, bool, bool, TdisplayAppearance, bool, bool, bool)
  TidleRefreshCmd::refresh(T3dView*, TdisplayAppearance, bool, bool, bool, bool, bool, bool)
  TidleRefreshCmd::doIt(Tevent const&)
  TeventHandler::doIdles()
  /mnt/rez/softwares/binaries/maya/2023.2/maya2023/lib/libExtensionLayer.so(+0x35187d) [0x7fcbfdf6687d]
  QObject::event(QEvent*)
  QApplicationPrivate::notify_helper(QObject*, QEvent*)

**Comment by santosg87** [comment]
this is fixed by #2698
